src/NzbDrone.Core/Notifications/Mailgun/Mailgun.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Core.Tv;
+
+namespace NzbDrone.Core.Notifications.MailGun
+{
+    public class MailGun : NotificationBase<MailGunSettings>
+    {
+        private readonly IMailgunProxy _proxy;
+        private readonly Logger _logger;
+        
+        public MailGun(IMailgunProxy proxy, Logger logger)
+        {
+            _proxy = proxy;
+            _logger = logger;
+        }
+
+        public override string Name => "Mailgun";
+        public override string Link => "https://mailgun.com";
+
+        public override void OnGrab(GrabMessage grabMessage)
+        {
+            _proxy.SendNotification(EPISODE_GRABBED_TITLE, grabMessage.Message, Settings);
+        }
+
+        public override void OnDownload(DownloadMessage downloadMessage)
+        {
+            _proxy.SendNotification(EPISODE_DOWNLOADED_TITLE, downloadMessage.Message, Settings);
+        }
+
+        public override void OnHealthIssue(HealthCheck.HealthCheck healthCheckMessage)
+        {
+            _proxy.SendNotification(HEALTH_ISSUE_TITLE, healthCheckMessage.Message, Settings);
+        }
+
+
+        public override ValidationResult Test()
+        {
+            var failures = new List<ValidationFailure>();
+
+            try
+            {
+                const string title = "Test Notification";
+                const string body = "This is a test message from Sonarr, though MailGun.";
+                Settings.TestMode = true;
```

Comment:
Why force TestMode to true here? this makes it impossible to actually test if it works (because you don't get an email).

Reply:
You don't gen an email, you're correct but you do see the email in the Mailgun dashboard and see if the apikey for example is correct.